x-pack/plugins/ml/public/application/components/job_spaces_repair/job_spaces_repair_flyout.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { FC, useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiText,
+  EuiCallOut,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { ml } from '../../services/ml_api_service';
+import {
+  RepairSavedObjectResponse,
+  SavedObjectResult,
+} from '../../../../common/types/saved_objects';
+import { RepairList } from './repair_list';
+import { useToastNotificationService } from '../../services/toast_notification_service';
+
+interface Props {
+  onClose: () => void;
+}
+export const JobSpacesRepairFlyout: FC<Props> = ({ onClose }) => {
+  const { displayErrorToast, displaySuccessToast } = useToastNotificationService();
+  const [loading, setLoading] = useState(false);
+  const [repairable, setRepairable] = useState(false);
+  const [repairResp, setRepairResp] = useState<RepairSavedObjectResponse | null>(null);
+
+  async function loadRepairList(simulate: boolean = true) {
+    setLoading(true);
+    try {
+      const resp = await ml.savedObjects.repairSavedObjects(simulate);
+      setRepairResp(resp);
+
+      const count = Object.values(resp).reduce((acc, cur) => acc + Object.keys(cur).length, 0);
+      setRepairable(count > 0);
+      setLoading(false);
+      return resp;
+    } catch (error) {
+      // this shouldn't be hit as errors are returned per-repair task
+      // as part of the response
+      displayErrorToast(error);
+      setLoading(false);
+    }
+    return null;
+  }
+
+  useEffect(() => {
+    loadRepairList();
+  }, []);
+
+  async function repair() {
+    if (repairable) {
+      // perform the repair
+      const resp = await loadRepairList(false);
+      // check simulate the repair again to check that all
+      // items have been repaired.
+      await loadRepairList(true);
+
+      if (resp === null) {
+        return;
+      }
+      const { successCount, errorCount } = getResponseCounts(resp);
+      if (errorCount > 0) {
+        const title = i18n.translate('xpack.ml.management.repairSavedObjectsFlyout.repair.error', {
+          defaultMessage: 'Some jobs cannot be repaired.',
+        });
+        displayErrorToast(resp as any, title);
+        return;
+      }
+
+      displaySuccessToast(
+        i18n.translate('xpack.ml.management.repairSavedObjectsFlyout.repair.success', {
+          defaultMessage: '{successCount} {successCount, plural, one {job} other {jobs}} repaired',
+          values: { successCount },
+        })
+      );
+    }
+  }
+
+  return (
+    <>
+      <EuiFlyout maxWidth={600} onClose={onClose}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="m">
+            <h2>
+              <FormattedMessage
+                id="xpack.ml.management.repairSavedObjectsFlyout.headerLabel"
+                defaultMessage="Repair saved objects"
```

Comment:
Did you consider any alternative terms to 'Repair' for this functionality, such as 'Sync'? The word 'repair' could imply something is broken, which is not the case; it's just a case of needing to sync up the saved objects and jobs.

Reply:
This button could be renamed.
I think this would be more of a product decision, I'm happy to change the text if needed.
I think the `repair` endpoint could remain the same as it is not user facing. But if we decided that it needs to change then we should also change all relevant functions and files to remove the word "repair"

Reply:
My vote is to rename this to 'Sync' or 'Synchronize', and if we change the label on the button we should also change the endpoint to match in case it becomes more widely used by Kibana API users. This can all be done in a follow-up.